google-cloud-examples/src/main/java/com/google/cloud/examples/bigquery/snippets/JobSnippets.java

```diff
@@ -48,4 +53,60 @@ public boolean isDone() {
     }
     return true;
   }
+
+  /**
+   * Example usage of {@code waitFor()}.
+   */
+  // [TARGET waitFor(WaitForOption...)]
+  public boolean waitFor() {
+    try {
+      // [START waitFor]
+      Job completedJob = job.waitFor();
+      if (completedJob == null) {
+        // job no longer exists
+      } else if (completedJob.status().error() != null) {
+        // job failed, handle error
+      } else {
+        // job completed successfully
+      }
+      // [END waitFor]
+    } catch (InterruptedException e) {
+      return true;
+    } catch (TimeoutException e) {
+      // Timeouts shouldn't happen without a timeout option.
+      return false;
+    } catch (BigQueryException e) {
+      return true;
+    }
+    return true;
+  }
+
+  /**
+   * Example usage of {@code waitFor()} with checking period and timeout.
+   */
+  // [TARGET waitFor(WaitForOption...)]
+  public boolean waitForWithOptions() {
+    try {
+      // [START waitFor]
+      Job completedJob =
+          job.waitFor(
+              WaitForOption.checkEvery(1, TimeUnit.SECONDS),
+              WaitForOption.timeout(60, TimeUnit.SECONDS));
+      if (completedJob == null) {
+        // job no longer exists
+      } else if (completedJob.status().error() != null) {
+        // job failed, handle error
+      } else {
+        // job completed successfully
+      }
+      // [END waitFor]
+    } catch (InterruptedException e) {
+      return true;
```

Comment:
Same as above, in the `isDone()` snippet your return `false` for `InterruptedException`.